railties/lib/rails/command.rb

```diff
@@ -89,7 +84,9 @@ def root
       def printing_commands # :nodoc:
         lookup!
 
-        (subclasses - hidden_commands).flat_map(&:printing_commands)
+        visible_commands = (subclasses - hidden_commands).flat_map(&:printing_commands)
+
+        (visible_commands - COMMANDS_IN_USAGE).sort
```

Comment:
This won't work because `visible_commands` is an array of tuples (e.g. `[["command_name", "description"], ...]`), and `COMMANDS_IN_USAGE` is an array of names.

Also, though, I'm not sure if we should perform this filtering here.  It seems like a concern of `HelpCommand` (and of `HelpCommand#help_extended` specifically).  Or is there a reason we want to do this filtering centrally?

Reply:
Hmm, weird I moved COMMANDS_IN_USAGE in https://github.com/rails/rails/commit/f52b1ccd0d637cfb9911ddf3b118b7e2c5ad5b49 with the exact same reason 🙂 .
I'll fix this...

Reply:
I've added assertions to make sure the commands are filtered.